### ISSUE TITLE
keep file-explorer pane shown-status

### ISSUE THREAD

**Issue by wences-dc-uba-ar** [LukeGeneva/vscode-auto-collapse-explorer#9]
I like the idea of this extension but...
I usually work with the file-explorer pane hidden, and it is a bit annoying that when I change change the editing file (with ctr+tab) the files pane opens itself.
May be it can test if file-pane is shown before do its magic.
Thanks!

**Comment by nixsee** [comment]
I agree with this - it would be great if this didn't open the pane every time a file is switched.

**Comment by LukeGeneva** [comment]
Sorry, I've been unable to find a fix for this so far. The issue holding this back is an inability to work effectively with the Explorer through the VS Code API. If I'm able to find a way to do that, this can be easily fixed.

**Comment by maxfahl** [comment]
Love this extension, but this bug made me disable it. Also, I can't double click on an item since the menu get re-drawn when I single click.

**Comment by ProLoser** [comment]
So I also only want this extension to activate if the explorer panel is actually visible.

Poking around google I found the following:

https://code.visualstudio.com/api/references/when-clause-contexts specifically `explorerViewletVisible`

I don't know how to make VSCode extensions but perhaps this could be added as a precondition before executing the logic?

**Comment by joonaspaakko** [comment]
The biggest gripe I have with the extension is directly related. When I do a global search with multiple results, normally I can click through them pretty quickly, but with the extension activating the explorer panel on every tab open, I have to keep reopening the search panel in order to click the next search result.

**Comment by miranquil** [comment]
It is so sad that this issue cannot be solved for now...

**Comment by ProLoser** [comment]
Can someone locate an upstream bug in the vscode repo that we can link as a blocker to this? If not we should create one if you want to help progress this problem along.

**Comment by symbion6** [comment]
Everything everyone above said ^^
It's a great extension and thank you for creating it :-)
However, not taking into account the visible state of the File Explorer means I cannot use it in my workflow.
Might [this](https://github.com/microsoft/vscode/issues/48198#issuecomment-399908931) be a starting point for a solution?

**Comment by ProLoser** [comment]
@symbion6 yup, you should code it lol

**Comment by LukeGeneva** [comment]
Fixed in **v1.1.0**.

Switching files with the Explorer hidden no longer opens it. The extension now only
collapses and reveals while the Explorer is the visible sidebar view; with the
sidebar closed it does nothing.

VS Code still exposes no API for sidebar visibility
([microsoft/vscode#48198](https://github.com/microsoft/vscode/issues/48198),
[microsoft/vscode#255992](https://github.com/microsoft/vscode/issues/255992), both
closed as not planned) — thanks to everyone who tracked those down. The workaround
is that an extension *can* read the visibility of its own views, so v1.1.0 adds an
empty **Auto-Collapse Explorer** section to the Explorer and uses it as the signal.

One caveat: leave that section expanded. VS Code reports a collapsed section as not
visible, and there is no way to tell that apart from a closed sidebar, so collapsing
it turns the extension off. See the README.

**Comment by nickchomey** [comment]
@LukeGeneva Looks like it is working fairly well. A couple thoughts/questions:

1. It has a weird, somewhat annoying flicker - is that unavoidable?
2. is there any way to make the Auto-Collapse Explorer section smaller? It seems to hit a limit, probably imposed by VS Code. 
<img width="449" height="136" alt="Image" src="https://github.com/user-attachments/assets/f61d031e-58ae-408c-a5ab-4f9ed3573820" />

3. In a multi-root workspace, it doesn't collapse the roots themselves, just the subdirectories within them 

<img width="392" height="602" alt="Image" src="https://github.com/user-attachments/assets/a7738ace-c310-41d0-bc64-928398ac2de6" />
